krynn/krynn.php

```diff
@@ -0,0 +1,165 @@
+	unregister_hook('plugin_settings', 'addon/krynn/krynn.php', 'krynn_settings');
+	unregister_hook('plugin_settings_post', 'addon/krynn/krynn.php', 'krynn_settings_post');
+
+
+	logger("removed krynn");
+}
+
+
+
+function krynn_post_hook($a, &$item) {
+
+	/**
+	 *
+	 * An item was posted on the local system.
+	 * We are going to look for specific items:
+	 *      - A status post by a profile owner
+	 *      - The profile owner must have allowed our plugin
+	 *
+	 */
+
+	logger('krynn invoked');
+
+	if(! local_user())   /* non-zero if this is a logged in user of this system */
+		return;
+
+	if(local_user() != $item['uid'])    /* Does this person own the post? */
+		return;
+
+	if($item['parent'])   /* If the item has a parent, this is a comment or something else, not a status post. */
+		return;
+
+	/* Retrieve our personal config setting */
+
+	$active = get_pconfig(local_user(), 'krynn', 'enable');
+
+	if(! $active)
+		return;
+
+	/**
+	 *
+	 * OK, we're allowed to do our stuff.
+	 * Here's what we are going to do:
+	 * load the list of timezone names, and use that to generate a list of krynn locales.
+	 * Then we'll pick one of those at random and put it in the "location" field for the post.
+	 *
+	 */
+
+	$krynn = array('Ansalon','Abanasinia','Solace','Haven','Gateway','Qualinost','Ankatavaka','Pax Tharkas','Ergoth','Newsea','Straights of Schallsea','Plains of Dust','Tarsis','Barren Hills','Que Shu','Citadel of Light','Solinari','Hedge Maze','Tower of High Sorcery','Inn of the Last Home','Last Heroes Tomb','Academy of Sorcery','Gods Row','Temple of Majere','Temple of Kiri-Jolith','Temple of Mishakal','Temple of Zeboim,','The Trough','Sad Town','Xak Tsaroth','Zhaman','Skullcap','Saifhum','Karthay','Mithas','Kothas','Silver Dragon Mountain','Silvanesti');
+
+	$planet = array_rand($krynn,1);
+	$item['location'] = $krynn[$planet];
+
+	return;
+}
+
+
+
+
+/**
+ *
+ * Callback from the settings post function.
+ * $post contains the $_POST array.
+ * We will make sure we've got a valid user account
+ * and if so set our configuration setting for this person.
+ *
+ */
+
+function krynn_settings_post($a,$post) {
+	if(! local_user())
+		return;
+	if($_POST['krynn-submit'])
+		set_pconfig(local_user(),'krynn','enable',intval($_POST['krynn']));
+}
+
+
+/**
+ *
+ * Called from the Plugin Setting form. 
+ * Add our own settings info to the page.
+ *
+ */
+
+
+
+function krynn_settings(&$a,&$s) {
+
+	if(! local_user())
+		return;
+
+	/* Add our stylesheet to the page so we can make our settings look nice */
+
+	$a->page['htmlhead'] .= '<link rel="stylesheet"  type="text/css" href="' . $a->get_baseurl() . '/addon/krynn/krynn.css' . '" media="all" />' . "\r\n";
+
+	/* Get the current state of our config variable */
+
+	$enabled = get_pconfig(local_user(),'krynn','enable');
+
+	$checked = (($enabled) ? ' checked="checked" ' : '');
+
+	/* Add some HTML to the existing form */
+
+	$s .= '<div class="settings-block">';
```

Comment:
could you add some HTML code for the "minimized" settings header (see e.g. the [forumlist addon](https://github.com/friendica/friendica-addons/blob/master/forumlist/forumlist.php#L160)). Thhey are missing in your templlate addon as well I've just noticed looking over your code ;-)